src/main/java/uk/gov/hmcts/reform/userprofileapi/util/UserProfileValidator.java

```diff
@@ -117,4 +121,19 @@ static void validateUserProfileDataAndUserId(UpdateUserProfileData userProfileDa
             throw new RequiredFieldMissingException("No userId or roles in the request");
         }
     }
+
+    static void validateUserProfileDataAndFirstAndLastNames(UpdateUserProfileData userProfileData, String userId) {
+        if (null == userProfileData) {
+            throw new RequiredFieldMissingException("No Request Body in the request");
+
+        } else if (StringUtils.isBlank(userId)) {
+            throw new RequiredFieldMissingException("No User ID present in the request");
+
+        } else if (!StringUtils.isBlank(userProfileData.getFirstName()) && !Pattern.matches(NAME_FORMAT_REGEX, userProfileData.getFirstName())) {
+            throw new InvalidRequest("First name must only consist of Letters aA - zZ and the following special characters ' and -");
+
+        } else if (!StringUtils.isBlank(userProfileData.getLastName()) && !Pattern.matches(NAME_FORMAT_REGEX, userProfileData.getLastName())) {
+            throw new InvalidRequest("Last Name must only consist of Letters aA - zZ and the following special characters ' and -");
+        }
+    }

```

Comment:
Multiple _else if_ statements used so we can provide an accurate and detailed error response

Reply:
please rewrite the logic like add one method and pass just string which returns boolean variable true or false ..

Reply:
The validator has been removed in this commit: https://github.com/hmcts/rd-user-profile-api/pull/255/commits/f9a2479bada9aa6855a7a54a5391e4ebd18dfbc5

Reviewer comment no longer applicable.